dbt/include/bigquery/macros/utils/get_columns_spec_ddl.sql

```diff
@@ -5,6 +5,16 @@
 {%- endmacro -%}
 
 {% macro bigquery__get_empty_schema_sql(columns) %}
+    {%- set col_err = [] -%}
+    {% for col in columns.values() %}
+      {%- if col['data_type'] is not defined -%}
+        {{ col_err.append(col['name']) }}
+      {%- endif -%}
+    {%- endfor -%}
+    {%- if (col_err | length) > 0 -%}
+      {{ exceptions.column_type_missing(column_names=col_err) }}
+    {%- endif -%}
```

Comment:
This PR introduces some unsatisfying duplication in the bigquery version of the `get_empty_schema_sql` macro. 

However, it's sensible to do this before the columns get nested to provide granular error messages as opposed to deferring it to the handling in `default__get_empty_schema_sql` once the columns are already nested and it is not possible to determine which nested field had a missing data_type value. 

Once the [nested handling of model contracts moves into core](https://github.com/dbt-labs/dbt-core/issues/7759), this duplication should be resolved and only present in the `default__get_empty_schema_sql` implementation.